src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1270,7 +1270,8 @@ def matches_service(self, service_name):
     def service_id(self):
         if self.daemon_type == 'rgw':
             v = self.daemon_id.split('.')
-            return '.'.join(v[0:2])
+            # excluding `host.random_string`
+            return '.'.join(v[0:-2])
```

Comment:
what happens if the host has a ``.`` in it?  might be better to do a ``.``-delimited prefix match in the caller.

Reply:
wdyt?

Reply:
@sebastian-philipp i think Sage proposed a solution. whose is "y" you are pending on?

Reply:
I think this is good to go now.

Reply:
i think this will still lead to a similar failure if the host is something like myhost.mydomain.com.  should be fixable with something like ``fun is_service(self, svc: str): bool``.  that can be fixed later since this is still an improvement, or we could require that host names have no ``.``'s.

Reply:
> i think this will still lead to a similar failure if the host is something like myhost.mydomain.com. should be fixable with something like `fun is_service(self, svc: str): bool`. that can be fixed later since this is still an improvement, or we could require that host names have no `.`'s.

DDs fro cephadm already have the hostname property set, which makes this function actually work pretty well. Except if hostname is embedded into the RGW Zone or something.

Reply:
https://github.com/ceph/ceph/pull/34778 deals with having the hostname embedded in the zone